File: 2023-2024/sourcecode/mod1/lab2/42DynamicFragmentBasicsJava/app/src/main/java/com/example/a42_dynamicfragmentbasics_java/FragmentB.java
package com.example.a42_dynamicfragmentbasics_java;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.a42_dynamicfragmentbasics_java.databinding.FragmentBBinding;

public class FragmentB extends Fragment {

    private FragmentBBinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentBBinding.inflate(inflater);
        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod1a/lab3/11JetpackNavigationBasicsKotlin/app/src/main/java/com/example/a11_jetpacknavigationbasics_kotlin/FragmentB.kt
package com.example.a11_jetpacknavigationbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.example.a11_jetpacknavigationbasics_kotlin.databinding.FragmentBBinding

class FragmentB : Fragment() {

    private lateinit var binding: FragmentBBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentBBinding.inflate(inflater)

        binding.textView.text = arguments?.getInt("value").toString()

        binding.fabB.setOnClickListener {
            val action = FragmentBDirections.actionFragmentBToFragmentA()
            Navigation.findNavController(requireView()).navigate(action)
        }

        return binding.root
    }
}